src/api/c++/z3++.h

```diff
@@ -168,16 +180,26 @@ namespace z3 {
             Z3_set_ast_print_mode(m_ctx, Z3_PRINT_SMTLIB2_COMPLIANT);
         }
 
-
-        context(context const & s);
-        context & operator=(context const & s);
+        context(context const &) = delete;
```

Comment:
using = delete is a good change.
This could go into separate pull request to merge.